mantis/acquisition/autotracker.py

```diff
@@ -330,10 +331,15 @@ def estimate_shift(self, ref_img: ArrayLike, mov_img: ArrayLike, **kwargs) -> np
         if not autofocus_method_func:
             raise ValueError(f'Unknown autofocus method: {self.tracking_method}')
 
-        shifts_zyx = autofocus_method_func(ref_img=ref_img, mov_img=mov_img, **kwargs)
+        shifts_zyx_pix = autofocus_method_func(ref_img=ref_img, mov_img=mov_img, **kwargs)
 
         # shifts_zyx in px to shifts_zyx in um
-        self.shifts_zyx = np.array(shifts_zyx) * self.scale
+        shifts_zyx_um = np.array(shifts_zyx_pix) * self.scale
+
+        # Limit the shifts_zyx, preserving the sign of the shift
+        self.shifts_zyx = np.sign(shifts_zyx_um) * np.minimum(np.abs(shifts_zyx_um), self.shift_limit)
```

Comment:
are we going for all or nothing strategy or if it's off in one dimension it's ok?

Reply:
`len(self.shift_limit) = 3`, here we are capping each dimension to its max allowed value